### DISCUSSION TITLE
Classic BT with iPhone

### DISCUSSION THREAD

**Discussion by ivanvach** [mKeRix/room-assistant#835] [Q&A] [answered]
Introduction to room-assistant says I quote here

" room-assistant supports ... Bluetooth Classic, which allows you to track your phone even without a special app running. That way you can track iOS devices..."

However further in the documentation iPhone disappears and iWatch only is mentioned. Furthermore, MAC is required in the settings of BT Classic for tracking. It is well known that iPhone changes BT MAC randomly for privacy reasons so that it is impossible to be used for tracking.

Is it possible to track iPhone without special app running ? Can you track by iPhone name only? If yes, than please provide an example of the settings as it seems now that MAC address is in the core even if a name is to be used. 

Thanks

**Comment by AML225** [comment, marked as answer]
Classic BT worked quite well for me with two iPhones in my household. I set it up like any other BT device using the BT MAC address in Settings>General>About. That address doesn't change.

**Reply by mKeRix** [reply]
As @AML225 said, BT Classic always worked for iPhones as well. The hardware MAC is set and doesn't change, it's just the MAC that they advertise when you have the Bluetooth menu open that regularly changes. Classic is a request-response tracking system where specific requests are sent to the hardware MAC that you configure (and the device then responds), unlike BLE where we (mostly) just passively listen.

**Reply by ivanvach** [reply]
@mKeRix Thanks a lot. This makes sense. I will give it a try

**Reply by ivanvach** [reply]
Yes it does work !!! Thanks a lot for the eye opening response. I have a lot to learn about BT obviously.
It reports to MQTT every 10 seconds. Couldn't find how to change the reporting frequency. Can I make it 5 sec. for example?
Next thing to try is the API. I assume it will respond to the API.

**Reply by mKeRix** [reply]
`bluetoothClassic` has an `interval` setting ([docs](https://www.room-assistant.io/integrations/bluetooth-classic.html#settings)). Note that the overall update time is `interval * max(numInstances, numTrackedDevices)` since it operates on a round robin principle.

**Reply by ivanvach** [reply]
Thanks. This makes sense.

**Comment by ivanvach** [comment]
Many postings online suggest that after iOS 8 iPhone started randomizing advertised MAC addresses. Here is one reliable source confirming that https://appleinsider.com/articles/14/06/09/mac-address-randomization-joins-apples-heap-of-ios-8-privacy-improvements

**Reply by AML225** [reply]
Go ahead and try it, like I said I was using it reliably for months as recently as earlier this year.

**Reply by ivanvach** [reply]
I did. It works !!! Thanks. It is slow however takes 10 minutes to adjust distance from 20 to 0 when I moved the iPhone next to Pi.
